code/modules/mob/living/basic/minebots/minebot_upgrades.dm

```diff
@@ -0,0 +1,61 @@
+/// Minebot Melee Damage Upgrade
+/obj/item/mine_bot_upgrade
+	name = "minebot melee upgrade"
+	desc = "A minebot upgrade."
+	icon_state = "door_electronics"
+	icon = 'icons/obj/doors/door_assembly.dmi'
+	new_attack_chain = TRUE
+
+/obj/item/mine_bot_upgrade/interact_with_atom(atom/target, mob/living/user, list/modifiers)
+	var/mob/living/basic/mining_drone/minebot = target
+	if(!istype(minebot))
+		return ..()
+
+	upgrade_bot(minebot, user)
+	return ITEM_INTERACT_COMPLETE
+
+/obj/item/mine_bot_upgrade/proc/upgrade_bot(mob/living/basic/mining_drone/M, mob/user)
+	if(M.melee_damage_upper != initial(M.melee_damage_upper))
+		to_chat(user, "[M] already has a combat upgrade installed!")
```

Comment:
Add some spans to these to_chats please, probably just warning and info

Reply:
Done.